### PR TITLE
[PORT] VV editor expansions + some changes

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
# Friendly reminder for porters
* https://github.com/BeeStation/BeeStation-Hornet/pull/10844
* https://github.com/BeeStation/BeeStation-Hornet/pull/10885
  I made a mistake to `isappearance()` which doesn't have `isimage()` in the macro define.

## About The Pull Request
Ports VV editor up to date from TG
* https://github.com/tgstation/tgstation/pull/51451
* https://github.com/tgstation/tgstation/pull/58173
* https://github.com/tgstation/tgstation/pull/61302
* https://github.com/tgstation/tgstation/pull/62413
* https://github.com/tgstation/tgstation/pull/62982
* https://github.com/tgstation/tgstation/pull/64521
* https://github.com/tgstation/tgstation/pull/71090
* https://github.com/tgstation/tgstation/pull/71280
* https://github.com/tgstation/tgstation/pull/76091
* https://github.com/tgstation/tgstation/pull/78725
* https://github.com/tgstation/tgstation/pull/78924
* https://github.com/tgstation/tgstation/pull/78948
* https://github.com/tgstation/tgstation/pull/79154
* https://github.com/tgstation/tgstation/pull/79282

This PR included non-relevant PRs because of code stability. (i.e. list, callproc)

Not included:
* https://github.com/tgstation/tgstation/pull/63665
* https://github.com/tgstation/tgstation/pull/71110
* https://github.com/tgstation/tgstation/pull/77639

## Why It's Good For The Game
Our VV editor is damn old, and does weird behaviour. It makes me pissed off when it's not readible
but also introduces new VV editor features

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

---------------

image

image

image

VV editor is now readible more

-------------

image

New command: tagging

-----------

image

image

filter editor
the middle one is blurring
available from vv command

-------------

image

colour matrix editor
available from vv command

--------

image

new option

</details>

## Changelog
:cl: Various TG coders from multiple PRs (AnturK, Thunder12345, Fikou, Y0SH1M4S73R, Ryll/Shaps, MrStonedOne, LemonInTheDark, ninjanomnom) - port by EvilDragonfiend
fix: Datums and lists used as keys in assoc lists are now linked in vv
add: View variables can now accept input from a colour picker.
code: Changed resolve weakref vv option to TG version
admin: infinity option is added to vv classes
admin: Added the color matrix editor. It is available from a dropdown option in an atom's View Variables menu as well as when editing an existing color matrix.
admin: Added Datum Tagging, a new admin tool for quickly marking and referring back to various datums and atoms. You can tag atoms by ctrl+middle clicking them or using the right click context menu, or from their VV dropdown menu (especially for abstract datums that can't be clicked).
admin: The View-Tags menu can quickly be accessed by F9 default, or using the View-Tags button at the bottom of the Admin -> Game status panel section.
admin: Tagged datums can also be referenced for things like proccalls and VV var edits, similarly to Marked objects.
code: Fixup list helpers, remove listoflist footgun from generic list procs, remove duplicated procs
admin: VV for global vars will now load MUCH faster, in exchange lists are now perma contracted in that particular pane
admin: Appearance vars in VV now display instead of being left blank
fix: Fixes passing in list arguments in callproc
admin: VV can now display the contents of special byond lists like filters, or client.images
admin: VV on images now displays the image in the header
admin: VV can now display filters and includes their type
code: Precalculates special list var names and avoids incorrect vv targets
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
The hero of our time.

**Comment by JixS4v** [comment]
God on earth 🙏

**Review by Crossedfall — APPROVED** [review]

**Comment by Crossedfall** [comment]
Yolo

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -11,6 +11,15 @@
 
 #define isweakref(D) (istype(D, /datum/weakref))
 
+#define isimage(thing) (istype(thing, /image))
+
+GLOBAL_VAR_INIT(magic_appearance_detecting_image, new /image) // appearances are awful to detect safely, but this seems to be the best way ~ninjanomnom
+#define isappearance(thing) (!ispath(thing) && istype(GLOB.magic_appearance_detecting_image, thing))
```
Comment (EvilDragonfiend) [inline code/__DEFINES/is_helpers.dm:17]: Reminder: I ported this incorrectly. It should be
```dm
#define isappearance(thing) (!isimage(thing) && !ispath(thing) && istype(GLOB.magic_appearance_detecting_image, thing))
```

File: code/__DEFINES/is_helpers.dm [hunk 2]
```diff
@@ -11,6 +11,15 @@
...
+
+// The filters list has the same ref type id as a filter, but isnt one and also isnt a list, so we have to check if the thing has Cut() instead
+GLOBAL_VAR_INIT(refid_filter, TYPEID(filter(type="angular_blur")))
+#define isfilter(thing) (!hascall(thing, "Cut") && TYPEID(thing) == GLOB.refid_filter)
```
Comment (EvilDragonfiend) [inline code/__DEFINES/is_helpers.dm:21]: Reminder: `!islist(thing) && hascall(thing, "Cut") && TYPEID(thing) == GLOB.refid_filter` is correct. 
This `!hascall(thing, "Cut")` is wrong.
https://github.com/BeeStation/BeeStation-Hornet/pull/11507#discussion_r1760974456

**Pull request merged**